src/components/study/study-code-from-ide.tsx

```diff
@@ -0,0 +1,156 @@
+'use client'
+
+import { useQuery } from '@/common'
+import { OpenWorkspaceButton } from '@/components/study/open-workspace-button'
+import { listWorkspaceFilesAction } from '@/server/actions/workspace-files.actions'
+import {
+    ActionIcon,
+    Button,
+    Divider,
+    Group,
+    Paper,
+    Radio,
+    Table,
+    Text,
+    Title,
+} from '@mantine/core'
+import { useForm } from '@mantine/form'
+import { notifications } from '@mantine/notifications'
+import { DownloadSimpleIcon, TrashIcon } from '@phosphor-icons/react/dist/ssr'
+import { useEffect } from 'react'
+
+interface StudyCodeFromIDEProps {
+    studyId: string
+    onMainFileChange?: (fileName: string) => void
+    onFilesChange?: (files: string[]) => void
+}
+
+interface FormValues {
+    mainFile: string
+    files: string[]
+}
+
+export const StudyCodeFromIDE = ({ studyId, onMainFileChange, onFilesChange }: StudyCodeFromIDEProps) => {
+    const form = useForm<FormValues>({
+        initialValues: {
+            mainFile: '',
+            files: [],
+        },
+    })
+
+    const { data, isLoading, refetch, isFetching } = useQuery({
+        queryKey: ['workspace-files', studyId],
+        queryFn: async () => {
+            const result = await listWorkspaceFilesAction({ studyId })
+            if ('error' in result) {
+                throw new Error(typeof result.error === 'string' ? result.error : JSON.stringify(result.error))
+            }
+            return result
+        },
+    })
+
+    useEffect(() => {
+        if (data?.files && data.files.length > 0) {
+            const mainFile = data.suggestedMain ?? data.files[0]
+            form.setValues({ mainFile, files: data.files })
+            onMainFileChange?.(mainFile)
+            onFilesChange?.(data.files)
+        }
+    }, [data, form])
+
+    const handleMainFileChange = (fileName: string) => {
+        form.setFieldValue('mainFile', fileName)
+        onMainFileChange?.(fileName)
+    }
+
+    const handleRemoveFile = (fileName: string) => {
+        const newFiles = form.values.files.filter((f) => f !== fileName)
+        form.setFieldValue('files', newFiles)
+        onFilesChange?.(newFiles)
+
+        if (form.values.mainFile === fileName) {
+            const newMain = newFiles[0] ?? ''
+            form.setFieldValue('mainFile', newMain)
+            onMainFileChange?.(newMain)
+        }
+    }
+
+    const handleImportFromIDE = () => {
+        refetch()
```

Comment:
I'm missing how refetch will import the files from the IDE but I'll keep reading the PR